receiver/statsdreceiver/transport/packet_server.go

```diff
@@ -0,0 +1,131 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package transport // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/statsdreceiver/transport"
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"strings"
+
+	"go.opentelemetry.io/collector/consumer"
+	"go.uber.org/multierr"
+)
+
+type packetServer struct {
+	packetConn net.PacketConn
+	transport  Transport
+	address    string
+}
```

Comment:
Did you consider having two Server types `udpServer` and `unixGramServer` and creating one or the other depending on transport?

Reply:
Because of the issue od `addr` being `nil` only when using a Unix socket I saw more difficult to have that part separated than having everything on the same file.